shared/src/main/scala/mlscript/JSBackend.scala

```diff
@@ -47,12 +47,12 @@ class JSBackend {
       JSObjectPattern(fields map {
         case (Var(nme), (Var(als), _)) if nme === als => {
           val runtimeName = scope.declareParameter(nme);
-          if (JSField.isValidIdentifier(nme)) nme -> N
+          if (JSField.isValidIdentifier(nme) && !Symbol.isKeyword(nme)) nme -> N
```

Comment:
Why not move the `isKeyword` check inside `isValidIdentifier`, where it logically belongs? You should really think about how to minimize the possibility for mistakes and omissions by centralizing checking and sanitization...

Reply:
I also did some other refactors. I wonder if it is necessary to move all checks outside the codegen so that all names in codegen are fine.

Reply:
> Yes. We need to be consistent. Either we do all name sanitization in code-gen, or none at all. Since some checks are already done before, then better move everything there.

There are still some parts I am not sure about. Take this one for example:
```scala
if (JSField.isValidIdentifier(property.name)) {
  s".${property.name}"
} else {
  s"[${JSLit.makeStringLiteral(property.name)}]"
}
```

Should I move this? I think it is not a sanitization because it just assembles the code based on if the name is valid. However, it does do some checks.

Reply:
> 

I think I got it. (from another conversation below)